Nodejs/Product/Nodejs/SmartIndent.cs

```diff
@@ -46,19 +48,17 @@ sealed class SmartIndent : ISmartIndent {
         #region ISmartIndent Members
 
         public int? GetDesiredIndentation(VisualStudio.Text.ITextSnapshotLine line) {
-            var dte = (EnvDTE.DTE)NodejsPackage.GetGlobalService(typeof(EnvDTE.DTE));
-
-            var props = dte.get_Properties("TextEditor", "Node.js");
-            switch ((EnvDTE._vsIndentStyle)(int)props.Item("IndentStyle").Value) {
+            var langPrefs = NodejsPackage.GetNodejsLanguagePreferencesFromTypeScript();
+            switch ((EnvDTE._vsIndentStyle)langPrefs[0].IndentStyle) {
```

Comment:
I'll see if I can get rid of this gross casting now that we are not directly using EnvDTE.